crypto/fipsmodule/sha/asm/sha256-armv4.pl

```diff
@@ -491,7 +499,21 @@ ()
 	stmdb	sp!,{r4-r12,lr}
 
 	sub	$H,sp,#16*4+16
-	adr	$Ktbl,K256
+
+	@ K256 is just at the boundary of being easily referenced by an ADR from
+	@ this function. In Arm mode, when building with __ARM_ARCH=6, it does
+	@ not fit. By moving code around, we could make it fit, but this is too
+	@ fragile. For simplicity, just load the offset from
+	@ .LK256_shortcut_neon.
+	@
+	@ TODO(davidben): adrl would avoid a load, but clang-assembler does not
+	@ support it. We might be able to emulate it with a macro, but Android's
+	@ did not work when I tried it.
+	@ https://android.googlesource.com/platform/ndk/+/refs/heads/master/docs/ClangMigration.md#arm
+	ldr	$Ktbl,.LK256_shortcut_neon
+.LK256_add_neon:
+	add	$Ktbl,pc,$Ktbl
+
```

Comment:
In the upstream commit in https://github.com/google/boringssl/commit/12316ab445eef5317391a94bef733fa6ff175173#diff-8b24c6636ba47fbf2802841ee1b0fdc757c4e6a160166e0487e7a0ae6529bc05R618, the same handling is repeated again to remove the `sub` at line https://github.com/aws/aws-lc/blob/e310388b2c2bf39a499b6e3e0b799dfb3a217ed0/crypto/fipsmodule/sha/asm/sha256-armv4.pl#L647
Should we do it as well? I think it may be needed for other platforms that have the sha instructions but compile in thumb, right?

Reply:
Good catch. I'll snag that part as well.